site/source/docs/porting/networking.rst

```diff
@@ -29,7 +29,7 @@ Full POSIX Sockets over WebSocket Proxy Server
 
 Emscripten provides a native POSIX Sockets proxy server program, located in directory tools/websocket_to_posix_proxy/, that allows full POSIX Sockets API access from a web browser. This support works by proxying all POSIX Sockets API calls from the browser to the Emscripten POSIX Sockets proxy server (via transparent use of WebSockets API), and the proxy server then performs the native TCP/UDP calls on behalf of the page. This allows a web browser page to run full TCP & UDP connections, act as a server to accept incoming connections, and perform host name lookups and reverse lookups. Because all API calls are individually proxied, this support can be slow. This support is mostly useful for developing testing infrastructure and debugging.
 
-To use POSIX sockets proxying, link the application with flags "-lwebsocket.js -s PROXY_POSIX_SOCKETS=1 -s USE_PTHREADS=1 -s PROXY_TO_PTHREAD=1". That is, POSIX sockets proxying builds on top of the Emscripten WebSockets library, and requires multithreading and proxying the application main() to a pthread.
+To use POSIX sockets proxying, link the application with flags "-lwebsocket.js -s PROXY_POSIX_SOCKETS=1 -s USE_PTHREADS=1 -s PROXY_TO_PTHREAD=1 -s FORCE_FILESYSTEM=1". That is, POSIX sockets proxying builds on top of the Emscripten WebSockets library, and requires multithreading and proxying the application main() to a pthread. Without FORCE_FILESYSTEM=1, required syscalls may be optimized out, giving the runtime error "SYSCALLS.getStreamFromFD is not a function".
```

Comment:
Hmm, it is not intended to use `-s FORCE_FILESYSTEM=1` with sockets proxying - any code in the `SYSCALLS.*` will not be compatible with sockets proxying, but rather if you are getting such an error, I think it's due to some bits missing in the sockets proxying PR. Let's revert this doc piece back and look at the runtime error in more detail?